numpydoc/numpydoc.py

```diff
@@ -203,12 +203,28 @@ def mangle_signature(app, what, name, obj, options, sig, retann):
     if not hasattr(obj, '__doc__'):
         return
     doc = get_doc_object(obj, config={'show_class_members': False})
-    sig = doc['Signature'] or getattr(obj, '__text_signature__', None)
+    sig = (doc['Signature'] or
+           _clean_text_signature(getattr(obj, '__text_signature__', None)))
     if sig:
         sig = re.sub("^[^(]*", "", sig)
         return sig, ''
 
 
+def _clean_text_signature(sig):
+    if sig is None:
+        return None
+    start_pattern = re.compile(r"^[^(]*\(*")
+    start, end = start_pattern.search(sig).span()
+    start_sig = sig[start:end]
+    sig = sig[end:]
+    sig = re.sub(r"\)", "", sig)
+    params = sig.replace(' ', '').split(',')
```

Comment:
this is too aggressive. Why do we need to replace all spaces? What if the value is rendered with spaces in it?

Doesn't
```suggestion
    params = sig.split(', ')
```
suffice?

Reply:
Fair point. I wasn't considering default values. Additionally, I think it should be pretty safe to assume that you'd rarely see something like `func(param1,param2)` (with no space) or `func(param1,  param2)` with too many spaces.